dev-docs/e2e/bare-metal-runner.md

```diff
@@ -1,5 +1,230 @@
+Feb 27 19:32:31 hetzner-ax162-snp kernel: SEV-SNP: Reserving start/end of RMP table on a 2MB boundary [0x0000000075a00000]
+Feb 27 19:32:32 hetzner-ax162-snp kernel: ccp 0000:09:00.5: sev enabled
+Feb 27 19:33:13 hetzner-ax162-snp kernel: ccp 0000:09:00.5: SEV API:1.55 build:32
+Feb 27 19:33:13 hetzner-ax162-snp kernel: ccp 0000:09:00.5: SEV-SNP API:1.55 build:32
+Feb 27 19:33:13 hetzner-ax162-snp kernel: kvm_amd: SEV enabled (ASIDs 100 - 1006)
+Feb 27 19:33:13 hetzner-ax162-snp kernel: kvm_amd: SEV-ES enabled (ASIDs 1 - 99)
+Feb 27 19:33:13 hetzner-ax162-snp kernel: kvm_amd: SEV-SNP enabled (ASIDs 1 - 99)
+```
+
+Install `docker` so that the docker login step in the CI succeeds.
+On Ubuntu, add it to the apt repositories (see https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository).
+```bash
+# Add Docker's official GPG key:
+sudo apt-get update
+sudo apt-get install ca-certificates curl
+sudo install -m 0755 -d /etc/apt/keyrings
+sudo curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
+sudo chmod a+r /etc/apt/keyrings/docker.asc
+
+# Add the repository to Apt sources:
+echo \
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu \
+  $(. /etc/os-release && echo "${UBUNTU_CODENAME:-$VERSION_CODENAME}") stable" | \
+  sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+sudo apt-get update
+
+# Download docker package
+sudo apt-get install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
+```
+
+Install k9s (optional)
+```bash
+pushd $(mktemp -d)
+wget https://github.com/derailed/k9s/releases/latest/download/k9s_Linux_amd64.tar.gz
+tar xvf k9s_Linux_amd64.tar.gz
+mv k9s /usr/local/bin/k9s
+popd
+```
+
+Install helm
+```bash
+pushd $(mktemp -d)
+wget https://get.helm.sh/helm-v3.17.1-linux-amd64.tar.gz
+tar xvf helm-v3.17.1-linux-amd64.tar.gz
+mv linux-amd64/helm /usr/local/bin/helm
+popd
+```
+
+Install K3s
+```bash
+curl -sfL https://get.k3s.io | sh -
+```
+
+Install Longhorn into K3s
+```bash
+helm repo add longhorn https://charts.longhorn.io
+helm repo update
+helm install longhorn longhorn/longhorn --namespace longhorn-system --create-namespace
+
+```
+
+## Networking
+Add the device to the tailscale network.
+For this you have to have admin privileges, if you don't see the
+overview of machines when visiting
+https://login.tailscale.com/admin/machines, notify another engineer.
+
+On this page, click "Add device". In the settings add the "gpu-cluster"
+label. This is needed since all engineers have the"gpu-cluster-admin"
+role, which allows them to ssh into all devices that have the gpu-cluster
+tag.
+Follow the other instructions on the tailscale website to add the device
+and exectue the given script on the machine.
+After the installation, execute:
+```bash
+sudo tailscale up --ssh
```

Comment:
How does this interact with the system sshd?

Reply:
> With Tailscale SSH, Tailscale takes over port 22 for SSH connections incoming from the Tailscale network. Tailscale will authenticate and encrypt the connection over WireGuard, using Tailscale node keys. The SSH client and server will still create an encrypted SSH connection, but it will not be further authenticated.

Thanks, that was what I was looking for.